### PR TITLE
CAL-69: created docs module within alliance

### PR DESCRIPTION
**Pull request by ricklarsen**
#### What does this PR do?
- Creates documentation module with alliance.
- draft of Managing NSILI app documentation

#### Who is reviewing it?
@troymohl 
@lcrosenbu 
@jlcsmith 

#### What are the relevant tickets?

https://codice.atlassian.net/browse/CAL-69

#### Screenshots (if appropriate)
n/a
#### Checklist:
- [x] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

**Comment by troymohl** [comment]
Since this work was started things have changed a little. 
NSILI Corba ORB has the following properties that used to be part of NSILI Endpoint:
-Corba Listen Port
-Corba Timeout

NSILI Endpoint has these properties:
-Max Number Returned Results
-Standing Query Update Rate
-Maximum Number Pending Results
-Sources to Query
 --> Source to Query -> add by clicking the plus and typing in the source name the same way it appears in the DDF Catalog - Sources tab in the configuration UI

File: distribution/docs/src/main/resources/_contents/_nsili/managing-nsili-contents.adoc [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+=== Prerequisites for ${alliance-nsili}
+
+The following apps must be installed with/before ${alliance-nsili}:
+
+* <<Managing ${ddf-spatial}, ${ddf-spatial}>>
+
+=== Installing ${alliance-nsili}
+
+This app is installed by default with standard installation.
```
Comment (lcrosenbu) [inline distribution/docs/src/main/resources/_contents/_nsili/managing-nsili-contents.adoc]: Change "This app..." to "The Alliance NSILI app..." to have the text independent of the heading.

File: distribution/docs/src/main/resources/_contents/_nsili/managing-nsili-contents.adoc [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+|NSILI_Federated_Source
+|NSILI Federated Source
+
+|<<NSILI_Connected_Source, NSILI Connected Source>>
+|NSILI_Connected_Source
+|NSILI Connected Source
+
+|<<org.codice.alliance.nsili.endpoint, NSILI Endpoint Configuration>>
+|org.codice.alliance.nsili.endpoint
+|
```
Comment (lcrosenbu) [inline distribution/docs/src/main/resources/_contents/_nsili/managing-nsili-contents.adoc]: Missing description.

File: distribution/docs/src/main/resources/_contents/_nsili/managing-nsili-contents.adoc [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+|Type
+|Description
+|Default Value
+|Required
+
+|Corba Listen Port
+|corbaPort
+|Corba Listen Port
+|Integer
+|20002
```
Comment (lcrosenbu) [inline distribution/docs/src/main/resources/_contents/_nsili/managing-nsili-contents.adoc]: This recently changed to 2809.

File: distribution/docs/src/main/resources/_contents/_nsili/managing-nsili-contents.adoc (hunk 1 above)
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_nsili/managing-nsili-contents.adoc]: fixed.

File: distribution/docs/src/main/resources/_contents/_nsili/managing-nsili-contents.adoc (hunk 2 above)
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_nsili/managing-nsili-contents.adoc]: fixed.

File: distribution/docs/src/main/resources/_contents/_nsili/managing-nsili-contents.adoc (hunk 3 above)
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_nsili/managing-nsili-contents.adoc]: fixed.

**Comment by peterhuffer** [comment]
The NSILI Federated and Connected Sources also have the following fields
-HTTP/S Timeout
-Additional Query Params

**Comment by jlcsmith** [comment]
looks like a rebase is needed

File: pom.xml [hunk 4]
```diff
@@ -62,6 +67,31 @@
...
+        <ddf-branding-expanded>Distributed Data Framework</ddf-branding-expanded>
+        <admin-console>Admin Console</admin-console>
+        <command-console>Command Console</command-console>
+        <!--Needed for documentation formatting-->
+        <at-symbol>@</at-symbol>
+        <variable-prefix>${</variable-prefix>
+        <image-width>500</image-width>
+        <!-- Individual application names -->
+        <ddf-admin>DDF Admin</ddf-admin>
+        <ddf-catalog>DDF Catalog</ddf-catalog>
```
Comment (jlcsmith) [inline pom.xml]: Aren't we trying to remove the distribution name from the application names?  So shouldn't this just be "Admin, Catalog, Platform etc?"

File: pom.xml [hunk 5]
```diff
@@ -62,6 +67,31 @@
...
+        <ddf-content>DDF Content</ddf-content>
```
Comment (jlcsmith) [inline pom.xml]: shouldn't the content app be removed?

Comment (ricklarsen) [inline pom.xml]: Yes. fixed.

File: pom.xml (hunk 4 above)
Comment (ricklarsen) [inline pom.xml]: Eventually, yes, but I don't think that change has happened yet. I think the docs should just reflect what the user will see.

File: distribution/docs/src/main/resources/_contents/_imaging/managing-imaging-contents.adoc [hunk 6]
```diff
@@ -0,0 +1,44 @@
+
+The ${alliance-imaging} Application provides support for ingesting and searching for NITF products.
+
+=== Prerequisites for ${alliance-imaging}
+
+None.
+
+=== Installing ${alliance-imaging}
+
+This application is installed by default on a standard ${branding} installation.
```
Comment (lcrosenbu) [inline distribution/docs/src/main/resources/_contents/_imaging/managing-imaging-contents.adoc]: Change "This app..." to "The Alliance NSILI app..." to have the text independent of the heading.

**Comment by jlcsmith** [comment]
I'm seeing some parameters not getting replaces in the generated html/pdf files - specifically ${secure_url}

**Comment by jlcsmith** [comment]
Other than the above comment, it looks good to me 👍

**Comment by ricklarsen** [comment]
Added properties for secure/public url replacement.

File: distribution/docs/pom.xml [hunk 7]
```diff
@@ -0,0 +1,414 @@
...
+                    <activeByDefault>true</activeByDefault>
+                </activation>
+                <properties>
+                    <asciidoctor.maven.plugin.version>1.5.2</asciidoctor.maven.plugin.version>
+                    <packaging>pom</packaging>
+                    <module>docs</module>
+                    <skipTests>false</skipTests>
+                </properties>
+                <repositories>
+                    <repository>
```
Comment (shaundmorris) [inline distribution/docs/pom.xml]: @ricklarsen do you know if we can use maven central for Ruby gems?

File: distribution/docs/src/main/resources/documentation.adoc [hunk 8]
```diff
@@ -0,0 +1,283 @@
...
+<<<
+
+== Integrating ${ddf-ui}
+[small]#Version: ${ddf.version}#
+
+include::${project.build.directory}/doc-contents/_ddf/docs-${ddf.version}/_ui-contents/integrating-ui-contents.adoc[]
+
+<<<
+////
+TODO: imaging documentation
```
Comment (shaundmorris) [inline distribution/docs/src/main/resources/documentation.adoc]: @ricklarsen do we have tickets to replace these TODOs?

File: distribution/docs/pom.xml (hunk 7 above)
Comment (ricklarsen) [inline distribution/docs/pom.xml]: It _looks_ doable, as `asciidoctor-diagram` appears to be on maven central now.

File: distribution/docs/src/main/resources/documentation.adoc (hunk 8 above)
Comment (ricklarsen) [inline distribution/docs/src/main/resources/documentation.adoc]: Just wrote some. Added.

**Comment by shaundmorris** [comment]
👍

**Pull request merged**